render/render.c

```diff
@@ -1107,6 +1108,7 @@ ProcRenderAddGlyphs(ClientPtr client)
 
                 if (!pDst) {
                     err = BadAlloc;
+                    FreePicture((void *) pSrc, 0);
```

Comment:
Do we also need `FreeScratchPixmapHeader(pSrcPix);` here?

Reply:
indeed.